api/project.py

```
@@ -0,0 +1,110 @@
+            raise Exception(f"missing path: {path}")
+
+        # adjust url
+        # 'git@github.com:FalkorDB/code_graph.git'
+        url  = Repo(path).remotes[0].url
+        url = url.replace("git@", "https://").replace(":", "/").replace(".git", "")
+
+        name = path.name
+
+        return cls(name, path, url)
+
+    def analyze_sources(self, ignore: Optional[List[str]] = []) -> Graph:
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)